### PR TITLE
Bug fix/search 357

### PR DESCRIPTION
**Pull request by gnusi**
### Scope & Purpose

Fix handling of offsets for multi-byte UTF8 characters.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -2582,8 +2583,55 @@ AqlValue functions::Substring(ExpressionContext* ctx, AstNode const&,
...
+      const auto c = irs::utf8_utils::next_checked(begin, end);
+
+      if (irs::utf8_utils::INVALID_CODE_POINT == c) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  if (!validate(subStr)) {
```
Comment (MBkkt) [inline arangod/Aql/Functions.cpp:2628]: Do we want to always make it?
Maybe add some unchecked options?

**Review by MBkkt — APPROVED** [review]
🤡

File: arangod/Aql/Functions.cpp (hunk 1 above)
Comment (gnusi) [reply arangod/Aql/Functions.cpp:2628]: I fear yes, because AQL doesn't support binary data now :-(

**Review by jsteemann — APPROVED** [review]
LGTM!

**Pull request merged**